### DISCUSSION TITLE
Pretrained model for age prediction basing on audio files?

### DISCUSSION THREAD

**Discussion by Marcophono2** [speechbrain/speechbrain#2559] [Q&A]
Hello!
I try to find a pretrained model to predict the speaker's age (german language). GPT finally (after 30 attempts) created a code which seems to be better than random but is for my test wav files always about 10 years to old. That would be okay if that is ALWAYS the case, so I just had to do age-10 but I think this is too unsafe. So my question is: Are there better solutions that the code below? The gender prediction is always "male", so it is not working but I have another working solution for the gender prediction. Thank you in advance!
Marc

audio_file = "/home/marc/Desktop/AI/whatsapp/audios/491777000001.wav"

import torchaudio
from speechbrain.pretrained import SpeakerRecognition
import torch

signal, fs = torchaudio.load(audio_file)

classifier = SpeakerRecognition.from_hparams(source="speechbrain/spkrec-xvect-voxceleb", savedir="tmp")

embeddings = classifier.encode_batch(signal)

def predict_age_gender(embedding):
    age = int(torch.sum(embedding) % 100)  # Dummy-Implementierung für Altersvorhersage
    gender = "male" if embedding.mean() > 0 else "female"  # Dummy-Implementierung für Geschlechtsvorhersage
    return age, gender

embedding = torch.mean(embeddings, dim=0)
age, gender = predict_age_gender(embedding)
print(f"Geschätztes Alter: {age}, Geschätztes Geschlecht: {gender}")`

**Comment by TParcollet** [comment]
Dear @Marcophono2, doing age prediction in this way will not work. The code generated by GPT makes no real sense. Embedding is a speaker embedding, which is an abstract and latent representation of "some" speaker attributes (even to this day, it's not 100% known what speaker embeddings do not contain). You would need a model trained to discriminate with age, or think about some unsupervised techniques to do this. GPT will not be able to solve this for you however.
